src/lib/pubsub/pubsub_macros.h

```diff
@@ -268,21 +268,32 @@
   }                                                             \
   EAT_SEMICOLON
 
+/**
+ * Add a fake use of the publish function for 'messagename', so that
+ * the compiler does not call it unused.
+ */
+#define DISPATCH__FAKE_USE_OF_PUBFN_(messagename)                       \
+  do {                                                                  \
+    if (0) {                                                            \
+      publish_fn__##messagename((msg_arg_type__##messagename) 0);       \
+    }                                                                   \
+  } while (0)
+
 /*
  * This macro is for internal use.  It backs DISPATCH_ADD_PUB*()
  */
 #define DISPATCH_ADD_PUB_(connector, channel, messagename, flags)       \
-  (                                                                     \
-    ((void)publish_fn__ ##messagename),                                 \
+  do {                                                                  \
+    DISPATCH__FAKE_USE_OF_PUBFN_(messagename);                          \
```

Comment:
Sorry, this doesn't actually seem to work at all:

```
../src/core/or/orconn_event.c:33:7: error: expected expression
  if (DISPATCH_ADD_PUB(connector, orconn, orconn_state) != 0)
      ^
../src/lib/pubsub/pubsub_macros.h:305:5: note: expanded from macro
      'DISPATCH_ADD_PUB'
    DISPATCH_ADD_PUB_(connector, channel, messagename, 0)
    ^
../src/lib/pubsub/pubsub_macros.h:286:3: note: expanded from macro
      'DISPATCH_ADD_PUB_'
  do {                                                                  \
  ^
../src/core/or/orconn_event.c:35:7: error: expected expression
  if (DISPATCH_ADD_PUB(connector, orconn, orconn_status) != 0)
      ^
../src/lib/pubsub/pubsub_macros.h:305:5: note: expanded from macro
      'DISPATCH_ADD_PUB'
    DISPATCH_ADD_PUB_(connector, channel, messagename, 0)
    ^
../src/lib/pubsub/pubsub_macros.h:286:3: note: expanded from macro
      'DISPATCH_ADD_PUB_'
  do {                                                                  \
  ^
../src/core/or/orconn_event.c:31:46: error: unused parameter 'connector'
      [-Werror,-Wunused-parameter]
orconn_add_pubsub(struct pubsub_connector_t *connector)
                                             ^
3 errors generated.
```

Among other things, `pubsub_add_pub_()` returns `int`. Maybe revert this for now?

Reply:
oh dear.  I've tried to revert and do a better fix in 26dcd7557ab1f3.